velox/exec/tests/SpillTest.cpp

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/exec/Spill.h"
+#include <gtest/gtest.h>
+#include <algorithm>
+#include <array>
+#include "velox/common/file/FileSystems.h"
+#include "velox/exec/tests/utils/TempDirectoryPath.h"
+#include "velox/serializers/PrestoSerializer.h"
+#include "velox/vector/tests/VectorTestBase.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::exec;
+
+class SpillTest : public testing::Test,
+                  public facebook::velox::test::VectorTestBase {
+ protected:
+  void SetUp() override {
+    mappedMemory_ = memory::MappedMemory::getInstance();
+    if (!isRegisteredVectorSerde()) {
+      facebook::velox::serializer::presto::PrestoVectorSerde::
+          registerVectorSerde();
+    }
+    filesystems::registerLocalFileSystem();
+  }
+
+  memory::MappedMemory* mappedMemory_;
+};
+
+TEST_F(SpillTest, spillState) {
+  auto tempDirectory = exec::test::TempDirectoryPath::create();
+  // We make a state that has 2 partitions, each with its own file
+  // list. We write 10 sorted vectors in each partition. The vectors
+  // have the ith element = i * 20 + sequence, where sequence is the
+  // sequence number of the vector in the partition. When read back,
+  // both partitions produce an ascending sequence of integers without
+  // gaps.
+  SpillState state(
```

Comment:
This test uses only one sorting key and has the same value for targetFileSize and targetBatchSize. Would it make sense to add another test for 2+ sorting keys including a variable-width key (string) and different targetFileSize and targetBatchSize? I'm seeing that targetBatchSize is not used. I wonder why is it in SpillState?

Also, the test doesn't have any payload, e.g. just the keys. Would it make sense to add some payload?